src/vanilla/store2.ts

```diff
@@ -610,6 +611,14 @@ export const createStore = (): Store => {
       dev4_override_method: (key, fn) => {
         ;(store as any)[key] = fn
       },
+      dev4_restore_atoms: (values) => {
+        const pendingPair = createPendingPair()
+        for (const [atom, value] of values) {
+          setAtomStateValueOrPromise(getAtomState(atom), value)
+          recomputeDependents(pendingPair, atom)
```

Comment:
Please add the primitive atom check which we already have it in `store.ts`.
```suggestion
          if (hasInitialValue(atom)) {
            setAtomStateValueOrPromise(getAtomState(atom), value)
            recomputeDependents(pendingPair, atom)
          }
```